pkg/channel/message_dispatcher.go

```diff
@@ -190,7 +191,7 @@ func (d *MessageDispatcherImpl) executeRequest(ctx context.Context, url *url.URL
 	}
 
 	if span.IsRecordingEvents() {
-		err = kncloudevents.WriteHTTPRequestWithAdditionalHeaders(ctx, message, req, additionalHeaders, kncloudevents.PopulateSpan(span))
+		err = kncloudevents.WriteHTTPRequestWithAdditionalHeaders(ctx, message, req, additionalHeaders, tracing.PopulateSpan(span, url.String()))
```

Comment:
Since you're keeping `kncloudevents.PopulateSpan` (which is fine, to avoid breaking downstream), can you add the deprecated comment to such function to signal users to use this new one in `tracing` module?

Reply:
The `kncloudevents.PopulateSpan` function & file was removed in the PR. However, I hadn't considered how this change might impact downstream. 
The only references to `PopulateSpan` were from `message_dispatcher.go`, which I assumed would be shipped together. Are there downstream uses cases that would be affected ?

Reply:
I honestly don't remember, try to remove the function and check if the downstream checks break :smile:

Reply:
Since this PR removes the function I was looking at the downstream checks for the PR. It looks like all but the Kafka downstream checks are passing. I am not sure if the [Kafka failure](https://github.com/knative/eventing/pull/4659/checks?check_run_id=1579319117#step:6:549) is related to the changes in the PR.

I also did a github search in [knative org](https://github.com/search?q=org%3Aknative+kncloudevents.PopulateSpan&type=code) and [knative-sandbox org](https://github.com/search?q=org%3Aknative-sandbox+kncloudevents.PopulateSpan&type=Code) and the only reference is message_dispatcher.go, so this change feels safe. 

Any additional things I could check ?

Reply:
Hmm, rebased off master just now and now the downstream kafak test is passing but the rabbit and nats ones are failing due to go mod issues.